### PR TITLE
feat(IDP-561): Table Actions Columns

### PR DESCRIPTION
**Pull request by Giogia**
### Description

Added actions columns to table component

##### Table

Added edit and delete columns, and the ability to provide custom actions

### Issue ticket number 

[IDP-561](https://makeitapp.atlassian.net/jira/software/c/projects/IDP/boards/1220?selectedIssue=IDP-561)

### Checklist

- [x] added the link to the Jira task
- [x] commit message and branch name follow conventions
- [x] tests are included
- [x] changes are accessible and documented from components stories
- [x] typings are updated or integrated accordingly with your changes
- [ ] all added components are exported from index file (if necessary)
- [x] all added files include Apache 2.0 license
- [x] you are not committing extraneous files or sensible data
- [x] the browser console does not have any logged errors
- [x] necessary labels have been applied to this pull request (enhancement, bug, ecc.)

### CODE REVIEW

File: src/components/Table/Table.Action.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <Button
+          hierarchy={isDanger ? Danger : Neutral}
+          icon={icon}
+          type={Ghost}
+          onClick={() => onClick(record, index)}
+        />
+      </div>
+    ),
+    align: ColumnAlignment.Right,
+    width: '0',
```
Comment (RiccardoCorona-MiaPlatform) [inline src/components/Table/Table.Action.tsx:52]: Why is it a string? 🤔

File: src/components/Table/Table.module.css [hunk 2]
```diff
@@ -38,3 +38,8 @@
     padding-bottom: var(--column-padding) !important;
   }
 }
+
+.action {
+  margin: calc(var(--spacing-margin-md, 12) * -1px);
```
Comment (RiccardoCorona-MiaPlatform) [inline src/components/Table/Table.module.css:43]: Why -1? 😧

Comment (Giogia) [reply src/components/Table/Table.module.css:43]: We need to constrain the button with a negative margin to hide the ripple effect around the button

File: src/components/Table/Table.Action.tsx (hunk 1 above)
Comment (Giogia) [reply src/components/Table/Table.Action.tsx:52]: 0 as number is not interpreted correctly, I can specify '0px' for clarity

**Review by RiccardoCorona-MiaPlatform — APPROVED** [review]

File: src/components/Table/Table.module.css (hunk 2 above)
Comment (Danielecina) [reply src/components/Table/Table.module.css:43]: lo mettiamo commentato sopra? così lo ricordiamo tra X o Comunque è visibile alle persone che guardano questo codice?

File: src/components/Table/Table.tsx [hunk 3]
```diff
@@ -45,13 +50,37 @@ export const Table = <RecordType extends GenericRecord>({
...
   size,
   scroll,
 }: TableProps<RecordType>): ReactElement => {
-  const tablePagination = useMemo(() => pagination !== false && ({ ...Table.pagination, ...pagination }), [pagination])
+  const theme = useTheme()
+  const iconSize = theme?.shape?.size?.lg as IconProps['size'] ?? 24
+
+  const tableColumns = useMemo(() => [
+    ...columns,
+    ...actions?.map(getAction) ?? [],
```
Comment (Danielecina) [inline src/components/Table/Table.tsx:66]: usiamo l'or

File: src/components/Table/Table.module.css [hunk 4]
```diff
@@ -38,3 +38,9 @@
     padding-bottom: var(--column-padding) !important;
   }
 }
+
+.action {
+  /* Remove the button ripple effect by adding negative margin and hidden overflow*/
```
Comment (RiccardoCorona-MiaPlatform) [inline src/components/Table/Table.module.css:43]:
```suggestion
  /* Removes the button ripple effect by adding a negative margin and hidden overflow */
```

File: src/components/Table/Table.props.ts [hunk 5]
```diff
@@ -73,6 +73,20 @@ export type TableProps<RecordType extends GenericRecord> = {
...
+   * When overriding a default action no field is required.
+   *
+   * action `object`:
+   *   - dataIndex*: The custom action identifier or, if "edit" or "delete", the default action to override. <br> `string`
+   *   - icon*: The icon to be displayed for the action. <br> `ReactNode`
+   *   - isDanger: Whether the action is a possibly dangerous action. <br> `boolean`
+   *   - isDisabled: Whether the action is disabled for a specific row. <br>  `(record: RecordType, index: number) => boolean` | `boolean`
+   *   - onClick*: The callback invoked when clicking the corresponding action button. <br>  `(record: RecordType, index: number) => void`
+   *   - title: Column displayed label for the action. <br> `ReactNode`
+   */
```
Comment (RiccardoCorona-MiaPlatform) [inline src/components/Table/Table.props.ts:87]:
```suggestion
   * The array of custom actions, or default action overrides, to be displayed in the table's last column.
   * When overriding a default action no field is required.
   *
   * action `object`:
   *   - dataIndex*: The custom action identifier or, if "edit" or "delete", the default action to override. <br> `string`
   *   - icon*: The icon to be displayed for the action. <br> `ReactNode`
   *   - isDanger: Whether the action is possibly dangerous. <br> `boolean`
   *   - isDisabled: Whether the action is disabled for a specific row. <br>  `(record: RecordType, index: number) => boolean` | `boolean`
   *   - onClick*: The callback invoked when clicking the corresponding action button. <br>  `(record: RecordType, index: number) => void`
   *   - title: Column displayed label for the action. <br> `ReactNode`
   */
```

**Review by RiccardoCorona-MiaPlatform — APPROVED** [review]

**Review by Danielecina — APPROVED** [review]

**Pull request merged**